src/lesson17/fruits/Box.java

```diff
@@ -0,0 +1,38 @@
+package lesson17.fruits;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Box<T extends Fruit> {
+    private List<T> fruits;
+
+    public Box() {
+        this.fruits = new ArrayList<>();
+    }
+
+    public void add(T fruit){
+        fruits.add(fruit);
+    }
+
+    public float getWeight(){
+        float sum = 0.0F;
+        for (T fruit : fruits){
+            sum += fruit.getWeight();
+        }
+        return sum;
+    }
+
+    public <T extends Fruit> boolean compare(Box<T> other){
```

Comment:
`T` уже зарезервировано за классом. Будет конфликт.
```suggestion
    public <V extends Fruit> boolean compare(Box<T> other){
```